frontend/src/context/anticipatoryActionStateSlice/index.ts

```diff
@@ -174,6 +174,93 @@ export const loadAAData = createAsyncThunk<
   );
 });
 
+interface CalculateMapRenderedDistrictsParams {
+  selectedDateData: AnticipatoryActionState['selectedDateData'];
+  categoryFilters: AnticipatoryActionState['categoryFilters'];
+  selectedWindow: AnticipatoryActionState['selectedWindow'];
+}
+
+function calculateMapRenderedDistricts({
+  selectedDateData,
+  categoryFilters,
+  selectedWindow,
+}: CalculateMapRenderedDistrictsParams) {
+  if (Object.entries(selectedDateData).length === 0) {
+    return { 'Window 1': {}, 'Window 2': {} };
+  }
+  const mapped = Object.entries(selectedDateData).map<
+    Pick<
+      AnticipatoryActionDataRow,
+      'district' | 'windows' | 'category' | 'phase'
+    >
+  >(([district, districtData]) => {
+    // eslint-disable-next-line fp/no-mutating-methods
+    const sortedData = [...districtData].sort((a, b) => {
+      const aOrder = AADataSeverityOrder(a.category, a.phase);
+      const bOrder = AADataSeverityOrder(b.category, b.phase);
+
+      if (aOrder > bOrder) {
+        return -1;
+      }
+      if (aOrder < bOrder) {
+        return 1;
+      }
+      return 0;
+    });
+
+    const filtered = sortedData.filter(x => categoryFilters[x.category]);
+
+    // let the first window layer to render empty district
```

Comment:
This is most often NA I think but we can come back to this later.
This process also does not take into account the fact that once SET is attained it stays available. Or am I missing something? Is there not a way to reuse what's calculated for `dataForRows` in the HomeTable?